src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -38,7 +38,10 @@ pub enum AbstractDataType {
     // }
     // ```
     Value,
-    // A series of zero or more values whose type is described by the nested `String` (e.g. a list)
+    // A series of zero or more values whose type is described by the nested `element_type`
+    // and sequence type is described by nested `sequence_type` (e.g. a list).
+    // If there is no `element` constraint present in schema type then `element_type` will be None.
+    // If there is no `type` constraint present in schema type then `sequence_type` will be None.
```

Comment:
Can the `sequence_type` ever be the name of user-defined type? If not, could/should we model this as an enum?

```rust
enum SequenceType {
  List,
  SExp,
}

Sequence {
  element_type: Option<String>,
  sequence: SequenceType
}
```

Given that we now require the `element_type` to be specified for sequences, should that still be an `Option<String>` instead of a `String`?

Reply:
The idea of using `SequenceType` enum is good as its only possibilities are list or sexp. 

For `element_type`, I use `Option<String>` so that whenever either just `type` constraint or just `element` constraint is specified it will set `element_type` to `None` but only when it gets both of these constraints it will put `Some(TYPE_NAME)`. So this can be used to throw an error when `element_type` is `None` at the end of reading a type definition for code gen. Does that make sense?